standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -493,4 +524,29 @@ public static String joinWithSeparator(Iterable<?> strings) {
   public static String[] getListFromSeparatedString(String commaSeparatedString) {
     return commaSeparatedString.split("\\s*" + TXN_WRITE_EVENT_FILE_SEPARATOR + "\\s*");
   }
+
+  private static Path getCmRoot(Path path) throws IOException {
```

Comment:
Need some synchronization to avoid multiple path being added for same EZ..does file system gives that gurantee ?